android/src/main/java/com/geektime/reactnativeonesignal/RNOneSignal.java

```diff
@@ -31,13 +32,31 @@
     public static final String NOTIFICATION_RECEIVED_INTENT_FILTER = "GTNotificationReceived";
 
     private ReactContext mReactContext;
+    private boolean oneSignalInitDone;
 
     public RNOneSignal(ReactApplicationContext reactContext) {
         super(reactContext);
         mReactContext = reactContext;
         mReactContext.addLifecycleEventListener(this);
+        initOneSignal();
+    }
+
+    // Initialize OneSignal only once when an Activity is available.
+    // React creates an instance of this class to late for OneSignal to get the current Activity
+    //   based on registerActivityLifecycleCallbacks it uses to listen for the first Activity.
+    // However it seems it is also to soon to call getCurrentActivity() from the reactContext as well.
+    // This will normally succeed when onHostResume fires instead.
+    private void initOneSignal() {
+        Activity activity = mReactContext.getCurrentActivity();
```

Comment:
@avishayil This compiles just fine, however I do so a possible race condition where `mReactContext` maybe `null` that your suggestion will fix. Updated it in the commit below.